packages/react-native/src/private/webapis/performance/PerformanceObserver.js

```diff
@@ -190,96 +135,66 @@ export class PerformanceObserver {
 
     this.#validateObserveOptions(options);
 
-    let requestedEntryTypes;
+    if (this.#nativeObserverHandle == null) {
+      this.#nativeObserverHandle = this.#createNativeObserver();
+    }
 
     if (options.entryTypes) {
       this.#type = 'multiple';
-      requestedEntryTypes = new Set(options.entryTypes);
-    } else {
+      NativePerformanceObserver.observe(this.#nativeObserverHandle, {
+        entryTypes: options.entryTypes.map(performanceEntryTypeToRaw),
+        type: undefined,
+        buffered: undefined,
+        durationThreshold:
+          options.type === 'event' ? options.durationThreshold : undefined,
+      });
+    } else if (options.type) {
       this.#type = 'single';
-      requestedEntryTypes = new Set([options.type]);
+      NativePerformanceObserver.observe(this.#nativeObserverHandle, {
+        entryTypes: undefined,
+        type: performanceEntryTypeToRaw(options.type),
+        buffered: options.buffered,
+        durationThreshold:
+          options.type === 'event' ? options.durationThreshold : undefined,
+      });
     }
+  }
 
-    // The same observer may receive multiple calls to "observe", so we need
-    // to check what is new on this call vs. previous ones.
-    const currentEntryTypes = registeredObservers.get(this)?.entryTypes;
-    const nextEntryTypes = currentEntryTypes
-      ? union(requestedEntryTypes, currentEntryTypes)
-      : requestedEntryTypes;
-
-    // This `observe` call is a no-op because there are no new things to observe.
-    if (currentEntryTypes && currentEntryTypes.size === nextEntryTypes.size) {
+  disconnect(): void {
+    if (!NativePerformanceObserver) {
+      warnNoNativePerformanceObserver();
       return;
     }
 
-    registeredObservers.set(this, {
-      callback: this.#callback,
-      durationThreshold:
-        options.type === 'event' ? options.durationThreshold : undefined,
-      entryTypes: nextEntryTypes,
-    });
-
-    if (!isOnPerformanceEntryCallbackSet) {
-      NativePerformanceObserver.setOnPerformanceEntryCallback(
-        onPerformanceEntry,
-      );
-      isOnPerformanceEntryCallbackSet = true;
+    if (this.#nativeObserverHandle != null) {
```

Comment:
Should this be the opposite? If it's null?

Reply:
It should 👀